lib/encoding/yaml/parser.toit

```diff
@@ -0,0 +1,1689 @@
+    return null
+
+  c-non-specific-tag -> string?:
+    if match-char C-TAG_: return "!"
+    return null
+
+  c-ns-anchor-property -> string?:
+    try-parse:
+      if match-char C-ANCHOR_:
+        if res := ns-anchor-name:
+          return res
+    return null
+
+  ns-anchor-name -> string?:
+    mark := mark
+    try-parse:
+      if (repeat --at-least-one: ns-anchor-char):
+        return string-since mark
+    return null
+
+  c-l-plus-literal n/int -> string?:
+    try-parse:
+      if match-char C-LITERAL_:
+        if t := c-b-block-header:
+          spaces := find-leading-spaces-on-first-non-empty-line
+          if res := l-literal-content spaces - t[1] t[0]:
+            return res.join ""
+    return null
+
+  c-l-plus-folded n/int -> string?:
+    try-parse:
+      if match-char C-FOLDED_:
+        t := c-b-block-header
+        spaces := find-leading-spaces-on-first-non-empty-line
+        if res := l-folded-content spaces - t[1] t[0]:
+          return res.join ""
+    return null
+
+  c-b-block-header -> List?:
+    try-parse:
+      indent-char := match-range '1' '9'
+      chomp-char := match-chars { '-', '+' }
+      if not indent-char: indent-char = match-range '1' '9'
+      if s-b-comment:
+        chomp := CLIP_
+        if chomp-char == '-': chomp = STRIP_
+        else if chomp-char == '+': chomp = KEEP_
+
+        indent := 0
+        if indent-char: indent = indent-char - '0'
+
+        return [ chomp, indent ]
+    return null
+
+  l-literal-content n/int t/int -> List?:
+    try-parse:
+      content := optional --or-null:
+        res/List? := null
+        if first := l-nb-literal-text n:
+          next := repeat: b-nb-literal-next n
+          if chomped-last := b-chomped-last t:
+            res = flatten-list_ [first, next, chomped-last]
+        res
+      if chomped-empty := l-chomped-empty n t:
+        if content: return flatten-list_ [content, chomped-empty]
+        else: return chomped-empty
+    return null
+
+  l-folded-content n/int t/int -> List?:
+    try-parse:
+      content := optional --or-null:
+        lines/List? := null
+        if tmp := l-nb-diff-lines n:
+          if chomped-last := b-chomped-last t:
+            lines = flatten-list_ [tmp, chomped-last]
+        lines
+
+      if chomped-empty := l-chomped-empty n t:
+        if content: return flatten-list_ [content, chomped-empty]
+        else: return chomped-empty
+    return null
+
+  b-chomped-last t/int -> string?:
+    if t == STRIP_: if b-non-content: return ""
+    if b-as-line-feed: return "\n"
+    return null
+
+  l-chomped-empty n/int t/int -> List?:
+    if t == KEEP_: return l-keep-empty n
+    return l-strip-empty n
+
+  l-keep-empty n/int -> List?:
+    empty-lines := repeat: l-empty n BLOCK-IN_
+    optional: l-trail-comments n
+    if empty-lines: return List empty-lines.size "\n"
+    return []
+
+  l-strip-empty n/int -> List?:
+    repeat: s-indent-less-or-equals n and (b-non-content or l-eof)
```

Comment:
Some rules depends on repeat to return null when not parsed. This would make it both --ignore-result and --as-null (or not setting --ignore-result) even though it should be ignored. Seems to be to many flags